build/build.sh

```diff
@@ -16,7 +16,7 @@ else
 fi
 cd "${BASEDIR}"
 
-VERSION=$(date -u +%Y-%m-%d)-$(bash scripts/git/commit.sh)
+VERSION=$(make release-tag)
```

Comment:
I don't think `make release-tag` is sufficient to replace what we had currently -- specifically, consider the situation where the user's working copy is dirty.  In that case, we'll get the same VERSION as when the working copy was clean.  While the things currently named "release tags" (which I recommended changing to "version tags" above) can only identify commits since they map to tags, the VERSION here maps to a working copy state.  One solution might be to make a singular "version tag" that uses `commit.sh` rather than using `git describe`, and "version tags" that identify a tag will just happen to never use the `-dirty` suffix.

Reply:
Many thanks for the review @BenjaminPelletier. Let's agree on this before I address the previous comments. I just realised that commit.sh was used everywhere and focused naively on the makefile. :/

1. Is the following reflecting your proposal? We would use as `version tag` either the current commit tag name if the working copy is clean and it is tagged with a name under the form of `v#.#.#`. Else `commit.sh` output ?
Examples:
a. Tagged and Clean: v1.0.0
b. Clean not tagged: 2b64628c
c. Tagged and Dirty: 2b64628c-dirty 
c. Dirty: 2b64628c-dirty 

2. According to you, which version should we pass to the binary which is currently expecting a semver for `version.tag` ? 
3. Do you think we should preserve the date in the image tag for non tagged versions ?

Reply:
The new version implements the outcome of our [discussion](https://github.com/interuss/dss/pull/659#issuecomment-1009268568)